portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -31,34 +31,37 @@
 
 /** Servlet for handling comments data */
 @WebServlet("/comments")
-public class DataServlet extends HttpServlet {
+public class CommentServlet extends HttpServlet {
 
-  /* Datastore data is represented by entities which have a kind and certain properties,
-   * and the constants below define the kind and property names for comment entities.
-   * Changing them would mean comments saved with a previous definition would not be displayed anymore. */
-  private static final String COMMENT_KIND = "Comment";
-  private static final String CONTENT_PROPERTY = "content";
-
-  /* This is the name of the input form which is defined in the html of the main page */
+  /* The name of the input form which is defined in the html of the main page. */
   private static final String INPUT_FORM = "comment-input";
 
-  /* This specifies what URL the client is redirected to after a POST request. */
+  /* Specifies what URL the client is redirected to after a POST request. */
   private static final String REDIRECT_URL = "/index.html";
 
+  /* The query string parameter for the number of comments to be displayed as defined in the script. */
+  private static final String COMMENT_NUMBER_PARAM = "number";
+
   /**
    * Loads and returns comments from the datastore database.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query(COMMENT_KIND);
+    int commentNumber = Integer.parseInt(request.getParameter(COMMENT_NUMBER_PARAM));
+    Query query = new Query(CommentEntity.KIND.getLabel());
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     List<String> comments = new ArrayList<>();
+    int count = 0;
```

Comment:
You can use FetchOptions to limit the number of comments, see https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions.html